libs/remix-ui/run-tab/src/lib/components/instanceContainerUI.tsx

```diff
@@ -1,22 +1,79 @@
 // eslint-disable-next-line no-use-before-define
 import { CustomTooltip } from '@remix-ui/helper'
-import React from 'react'
-import { FormattedMessage, useIntl } from 'react-intl'
+import React, {useEffect, useRef} from 'react'
+import { FormattedMessage } from 'react-intl'
 import { InstanceContainerProps } from '../types'
 import { UniversalDappUI } from './universalDappUI'
 
 export function InstanceContainerUI(props: InstanceContainerProps) {
   const { instanceList } = props.instances
+  const env = useRef()
+
+  useEffect(() => {
+    const fetchSavedContracts = async () => {
+      env.current = await props.plugin.call('blockchain', 'getProvider')
+      if(env.current && env.current === 'injected') {
+        const allSavedContracts = localStorage.getItem('savedContracts')
+        if (allSavedContracts) {
+          const savedContracts = JSON.parse(allSavedContracts)
+          const { network } = await props.plugin.call('blockchain', 'getCurrentNetworkStatus')
+          if (savedContracts[env.current] && savedContracts[env.current][network.id]) {
```

Comment:
The saved contracts should probably be independent of the current provider (metamask, external provider, ..)

Reply:
this is done